### PR TITLE
MXNet ONNX export - Dot op support

### PR DESCRIPTION
**Pull request by Roshrini**
## Description ##
Adding support for Dot operator for onnx export functionality.

@sandeep-krishnamurthy @zhreshold 
### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -435,6 +435,101 @@ def convert_pad(node, **kwargs):
...
+    node_inputs = node["inputs"]
+    name = node["name"]
+
+    input_a_idx = kwargs["index_lookup"][node_inputs[0][0]]
+    input_node_a = proc_nodes[input_a_idx].name
+    input_b_idx = kwargs["index_lookup"][node_inputs[1][0]]
+    input_node_b = proc_nodes[input_b_idx].name
+
+    # Getting the attributes and assigning default values.
+    if "attrs" in node:
```
Comment (zhreshold) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Can use two lines for these block:
trans_a = int(node.get('attrs', {}).get('transpose_a', 0))
trans_b = int(node.get('attrs', {}).get('transpose_b', 0))

I am not sure if the transpose_a is encoded as string '1' or 'True', if the latter, then it's not valid

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 2]
```diff
@@ -435,6 +435,101 @@ def convert_pad(node, **kwargs):
...
+        attrs = node["attrs"]
+        trans_a = int(attrs["transpose_a"])
+        trans_b = int(attrs["transpose_b"])
+    else:
+        trans_a = 0
+        trans_b = 0
+
+    op_name = "transpose" + str(kwargs["idx"])
+
+    if trans_a == 0 and trans_b == 0:
```
Comment (zhreshold) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: there are duplicate code block which can be reused by splitting the logic of trans_a and trans_b

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Thanks! Good catch. Seems like mxnet allows both "1" and "True", made changes accordingly.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 2 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Simplified

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 3]
```diff
@@ -435,6 +435,72 @@ def convert_pad(node, **kwargs):
...
 
+def create_helper_trans_node(op_name, input_node, node_name):
+    """create extra transpose node for dot operator"""
+    helper, _, _ = import_onnx_modules()
+
+    node_name = op_name + "_" + node_name
+    trans_node = helper.make_node(
+        'Transpose',
+        inputs=[input_node],
+        outputs=[op_name+"_a"],
```
Comment (zhreshold) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: why still have hardcoded "_a"?

**Comment by Roshrini** [comment]
Closing this PR as changes to the branch were not getting reflected.
Created new PR here- https://github.com/apache/incubator-mxnet/pull/11771

**Pull request closed without merging**